### PR TITLE
stats5

### PR DESCRIPTION
**Pull request by emirhandumann**

### CODE REVIEW

File: editor-components/stats/stats5/stats5.tsx [hunk 1]
```diff
@@ -58,38 +62,165 @@ class Stats5Page extends BaseStats {
...
       <div className={this.decorateCSS("container")}>
         <div className={this.decorateCSS("max-content")}>
           <div className={this.decorateCSS("bottom-child")}>
-            {this.castToObject<ICard[]>("card-content").map(
-              (cardData: any, indexCard: number) => (
-                <div key={indexCard} className={this.decorateCSS("card")}>
-                  <h4 className={this.decorateCSS("card-data-title")}>{cardData.title}</h4>
-                  <p className={this.decorateCSS("card-data-description")}>{cardData.description}</p>
+            {this.castToObject<CardData[]>("card-content").map(
+              (data: any, index: number) => (
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:207]: change data: CardData

File: editor-components/stats/stats5/stats5.tsx [hunk 2]
```diff
@@ -58,38 +62,165 @@ class Stats5Page extends BaseStats {
...
+                  key={index}
+                  className={`${this.decorateCSS("card")} ${this.getCardClasses(
+                    index,
+                    itemCountInRow
+                  )}`}
+                  style={{
+                    width: 90 / itemCountInRow + "%",
+                  }}
+                >
+                  <h4 className={this.decorateCSS("card-data-title")}>
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:218]: Remove divs when there is no input. Do this for all.

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
Please handle the changes.

File: editor-components/stats/stats5/stats5.tsx [hunk 3]
```diff
@@ -58,42 +62,168 @@ class Stats5Page extends BaseStats {
...
+                    style={{
+                      width: 90 / itemCountInRow + "%",
+                    }}
+                  >
+                    <h4 className={this.decorateCSS("card-data-title")}>
+                      {this.getComponentState(`number-${index}`)}
+                    </h4>
+                    <p className={this.decorateCSS("card-data-description")}>
+                      {data.CardDescription}
+                    </p>
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:221]: don't render p element if data.description is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/stats/stats5/stats5.tsx [hunk 4]
```diff
@@ -58,42 +62,168 @@ class Stats5Page extends BaseStats {
...
+                    className={`${this.decorateCSS(
+                      "card"
+                    )} ${this.getCardClasses(index, itemCountInRow)}`}
+                    style={{
+                      width: 90 / itemCountInRow + "%",
+                    }}
+                  >
+                    <h4 className={this.decorateCSS("card-data-title")}>
+                      {this.getComponentState(`number-${index}`)}
+                    </h4>
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:218]: don't render h4 if {this.getComponentState(`number-${index}`)} is empty
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/stats/stats5/stats5.tsx [hunk 5]
```diff
@@ -58,42 +62,168 @@ class Stats5Page extends BaseStats {
...
+    const totalCards = this.castToObject<CardData[]>("cardContent").length;
+    const isLastInRow =
+      (index + 1) % itemCountInRow === 0 || index === totalCards - 1;
+
+    return !isLastInRow ? this.decorateCSS("stick") : "";
+  }
+
   render() {
+    const itemCountInRow = this.getPropValue("itemCountInRow");
+    const hasCard = this.castToObject<CardData[]>("cardContent").length > 0;
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:198]:
```suggestion
    const cards = this.castToObject<CardData[]>("cardContent");
```

File: editor-components/stats/stats5/stats5.tsx [hunk 6]
```diff
@@ -58,42 +62,168 @@ class Stats5Page extends BaseStats {
...
-            {this.castToObject<ICard[]>("card-content").map(
-              (cardData: any, indexCard: number) => (
-                <div key={indexCard} className={this.decorateCSS("card")}>
-                  <h4 className={this.decorateCSS("card-data-title")}>{cardData.title}</h4>
-                  <p className={this.decorateCSS("card-data-description")}>{cardData.description}</p>
-                </div>
-              )
-            )}
-          </div>
+          {hasCard && (
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:203]:
```suggestion
          {cards.length > 0 && (
```

File: editor-components/stats/stats5/stats5.tsx [hunk 7]
```diff
@@ -58,42 +62,168 @@ class Stats5Page extends BaseStats {
...
+            <div className={this.decorateCSS("bottom-child")}>
+              {this.castToObject<CardData[]>("cardContent").map(
```
Comment (dogusmiuw) [inline editor-components/stats/stats5/stats5.tsx:205]:
```suggestion
              {cards.map(
```

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

**Review by Ayazelar — APPROVED** [review]

**Review by dogusmiuw — APPROVED** [review]

**Pull request merged**